src/editor/viewport/menu/MeshGenerator.js

```diff
@@ -8,15 +9,55 @@ export default class MeshGenerator{
             CIRCLE: 2,
             UVSPHERE: 3,
             ICOSPHERE: 4,
-            CYCLINDER: 5,
+            CYLINDER: 5,
             CONE: 6,
             TORUS: 7
         },
         CAMERA: 8,
         LIGHT: 9
     };
 
-    static create(objectType){
-        
+    constructor(viewport){
+        this.viewport = viewport;
+        this.cursorPoint = new Vector3(0, 0, 0);
+    }
+
+    create(objectType){
+        var geometry, material;
+        material = new THREE.MeshBasicMaterial({color:0x8e9091});
+        switch(objectType){
+            case OBJECT_TYPE.MESH.PLANE:
+                geometry = new THREE.PlaneGeometry(1, 1);
+                material = new THREE.MeshBasicMaterial({color:0x8e9091 , side: THREE.DoubleSide});
+                break;
+            case OBJECT_TYPE.MESH.CUBE:
+                geometry = new THREE.BoxGeometry(1, 1, 1);
+                break;
+            case OBJECT_TYPE.MESH.CIRCLE:
+                geometry = new THREE.CircleGeometry(1, 10);
+                break;
+            case OBJECT_TYPE.MESH.UVSPHERE:
+                geometry = new THREE.SphereGeometry(1, 30, 30);
+                break;
+            case OBJECT_TYPE.MESH.ICOSPHERE:
+                geometry = new THREE.IcosahedronGeometry(1, 2);
+                break; 
+            case OBJECT_TYPE.MESH.CYLINDER:
+                geometry = new THREE.CylinderGeometry(1, 1, 5, 10, 10);
+                break; 
+            case OBJECT_TYPE.MESH.CONE:
+                geometry = new THREE.ConeGeometry(1, 5, 10, 10);
+                break; 
+            case OBJECT_TYPE.MESH.TORUS:
+                geometry = new THREE.TorusGeometry(10, 5, 10, 30);
+                break; 
+            case OBJECT_TYPE.CAMERA:
+                geometry = new THREE.PerspectiveCamera(75, 1920/1200, 0.1, 1000);
+                break;
+            case OBJECT_TYPE.LIGHT:
+                geometry = new THREE.Light();
+                break; 
+        }
```

Comment:
could have change the position to the cursorPoint

Reply:
I've missed that one